### PR TITLE
Update the scroll-animations IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The Spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2018-03-05 21:34:56
Finished: 2018-03-05 21:41:21


<h2>Failing Jobs</h2>
<ul>

<li>firefox:nightly</li>

<li>build_css</li>

<li>chrome:dev</li>

<li>safari:11.0</li>

<li>MicrosoftEdge:16.16299</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25288)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/349439098)

**Comment by lukebjerring** [comment]
@foolip - Unsure about OWNERS. Also might need a tentative on the test? And where should the test go?

**Comment by foolip** [comment]
@birtles @theres-waldo @manta12, you are the editors listed in https://wicg.github.io/scroll-animations/ and show up in https://github.com/WICG/scroll-animations/graphs/contributors, so I added you as the suggested reviewers in META.yml. Can you confirm that you are OK with this? @theres-waldo @manta12, you'll also need to accept the invite to the web-platform-tests org to make this work.

**Comment by foolip** [comment]
@lukebjerring this need not be tentative, even though https://wicg.github.io/scroll-animations/ says "Unofficial Proposal Draft" the IDL is still in the spec, and if it changes the test will. Tentative tests are rather for things which haven't been merged into *any* spec.

**Review by foolip — APPROVED** [review]
Tests LGTM, but should await suggested reviewer OK before merging.

**Comment by theres-waldo** [comment]
> @birtles @theres-waldo @manta12, you are the editors listed in https://wicg.github.io/scroll-animations/ and show up in https://github.com/WICG/scroll-animations/graphs/contributors, so I added you as the suggested reviewers in META.yml. Can you confirm that you are OK with this? 

I'm not entirely sure what's going on / what I'm supposed to be OK-ing.

The Scroll Animations spec is likely to undergo changes (possibly significant ones) before progressing further in the standards process. There are no web platform tests (or any other kind of tests) for it yet.

**Comment by lukebjerring** [comment]
@theres-waldo - details about the reason behind this PR are in the description - basically we're looking for someone to be able to eyeball changes to the IDL each time it auto-updates, and verify whether the idlharness.js test is covering the interface well enough.

**Comment by theres-waldo** [comment]
> we're looking for someone to be able to eyeball changes to the IDL each time it auto-updates

Ok, the three of us added as suggested reviewers are appropriate for that.

> and verify whether the idlharness.js test is covering the interface well enough

This part I don't entirely follow. Is the idea that something generates tests based on the IDL alone?

**Comment by lukebjerring** [comment]
Yes, [idlharness.js](https://web-platform-tests.org/writing-tests/idlharness.html) is a framework in WPT for automatically generating a test suite based on the IDL itself, and some instances/JS objects which implement that IDL. 

The latter - making JS objects which implement the IDL - is often fairly niche/esoteric knowledge, so having the write pair of eyeballs on it can help a lot.

**Comment by theres-waldo** [comment]
Ok, so in this case, is there some generated JS code that implements the `ScrollTimeline` interface? Where is that?

File: scroll-animations/idlharness.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      const idl = await fetch('/interfaces/scroll-animations.idl').then(r => r.text());
+      const web = await fetch('/interfaces/web-animations.idl').then(r => r.text());
+      const dom = await fetch('/interfaces/dom.idl').then(r => r.text());
+
+      const idl_array = new IdlArray();
+      idl_array.add_idls(idl);
+      idl_array.add_dependency_idls(web);
+      idl_array.add_dependency_idls(dom);
+      idl_array.add_objects({
+        ScrollTimeline: ['new ScrollTimeline()'],
```
Comment (lukebjerring) [inline scroll-animations/idlharness.html]: @theres-waldo - to clarify, the JS object is not auto-generated, it's just created inline, here.

File: scroll-animations/META.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+suggested_reviewers:
+  - birtles
+  - theres-waldo
+  - manta12
```
Comment (birtles) [inline scroll-animations/META.yml]: Mantaroh is less involved in this spec recently so I think it's not necessary to include him.

**Review by birtles — COMMENTED** [review]
Looks good with the updates to the reviewer list made.

File: scroll-animations/META.yml (hunk 2 above)
Comment (foolip) [reply scroll-animations/META.yml]: OK, I'll remove him again.

**Pull request merged**